### ISSUE TITLE
jffi ships with binaries requiring glibc 2.27

### ISSUE THREAD

**Issue by headius** [jruby/jruby#7695]
Original issue is here: https://github.com/jnr/jffi/issues/138

jffi's builds are done on a recent version of Debian that has a recent glibc. Unfortunately glibc does not produce dynamic bindings that are compatible with older versions, so the jffi stub fails to load. I attempted to make jffi builds use an older debian in CI, but it did not appear to link against an older glibc.

This should be dealt with for JRuby 9.4.3 but I'm not sure how best to do it.

**Comment by headius** [comment]
This is still stuck. We partially fixed it for Linux on x86_64, but the image we were using (debian:8) no longer works in CI and we have other platforms that still need this fixed. It's getting more and more difficult to build anything on these old Linux distributions.

**Comment by RTrampov** [comment]
Any plans to fix this in JRuby 9.3.x.x, since 9.3.10.0 and 9.3.11.0 are also affected?

**Comment by jsvd** [comment]
Just for reference, to facilitate reproduction, here's a 1 liner with docker, testing with centos/oraclelinux 7 and aarch64:

```
docker run -it --platform linux/aarch64 centos:7 sh -c "yum install java-11-openjdk-headless.aarch64 -y --nogpgcheck -q -e 0 && curl -s https://repo1.maven.org/maven2/org/jruby/jruby-dist/9.4.5.0/jruby-dist-9.4.5.0-bin.tar.gz | tar -zxf - && jruby-9.4.5.0/bin/jruby -Xnative.verbose -w -d --dev -e 'system \"date\"'"
```

**Comment by headius** [comment]
The complication here is that our cross-platform build on GHA is designed around Debian, so we'd need to make a special build based on CentOS (package names etc) to force this build against older glibc.

**Comment by headius** [comment]
There's some updates in jnr/jffi#138 and a build with newer libffi that may not force a dependency on newer glibc via the memfd_create function...

**Comment by headius** [comment]
Fixed by #8105
